libsolidity/parsing/Parser.cpp

```diff
@@ -972,19 +973,43 @@ ASTPointer<UsingForDirective> Parser::parseUsingDirective()
 	expectToken(Token::Using);
 
 	vector<ASTPointer<IdentifierPath>> functions;
+	vector<optional<Token>> operators;
 	bool const usesBraces = m_scanner->currentToken() == Token::LBrace;
 	if (usesBraces)
 	{
 		do
 		{
 			advance();
 			functions.emplace_back(parseIdentifierPath());
+			if (m_scanner->currentToken() == Token::As)
+			{
+				advance();
+				Token operator_ = m_scanner->currentToken();
+				if (!util::contains(userDefinableOperators, operator_))
+				{
+					parserError(
+						4403_error,
+						fmt::format(
+							"Not a user-definable operator: {}. Only the following operators can be user-defined: {}",
+							(!m_scanner->currentLiteral().empty() ? m_scanner->currentLiteral() : string(TokenTraits::toString(operator_))),
+							util::joinHumanReadable(userDefinableOperators | ranges::views::transform([](Token _t) { return string{TokenTraits::toString(_t)}; }))
+						)
+					);
+				}
+				operators.emplace_back(operator_);
+				advance();
+			}
+			else
+				operators.emplace_back();
```

Comment:
I'm assuming the empty emplace will default initialize an `optional` - so `nullopt` here, but I'm not entirely sure why this is done? I know you've mentioned in a [comment](https://github.com/ethereum/solidity/pull/13790/files#diff-3f7b7120872e5e747b692c9a8bd8310aaf6e4ec285ef300ec4d4d32617fb5f8aR703) that the size of operators and functions must match - does the same apply here?
In any case, if this is in fact the case, I would prefer for calls such as this to be `emplace_back(nullopt)`.

Reply:
> I know you've mentioned in a [comment](https://github.com/ethereum/solidity/pull/13790/files#diff-3f7b7120872e5e747b692c9a8bd8310aaf6e4ec285ef300ec4d4d32617fb5f8aR703) that the size of operators and functions must match - does the same apply here?

Yes. We have two parallel vectors. One has functions and the other operators they implement (if the operator is `null`, the function is attached to the type instead). If we skip `null` values, you won't necessarily find the operator corresponding to n-th function at index `n`.

Technically, we could safely skip the final empty values in the `operators` array but having them there makes things more regular.